loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,98 @@
+ */
+export class KHR_mesh_instancing implements IGLTFLoaderExtension {
+    /**
+     * The name of this extension.
+     */
+    public readonly name = NAME;
+
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /** @hidden */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public loadNodeAsync(context: string, node: INode, assign: (babylonTransformNode: TransformNode) => void): Nullable<Promise<TransformNode>> {
+        return GLTFLoader.LoadExtensionAsync<IKHRMeshInstancing, TransformNode>(context, node, this.name, (extensionContext, extension) => {
+            return this._loader.loadNodeAsync(`#/nodes/${node.index}`, node, (babylonTransformNode) => {
+                const promises = new Array<Promise<any>>();
+                const attributeBuffers : { [name: string]: { buffer : Float32Array, accessor : IAccessor } } = {};
+                const loadAttribute = (attribute: string) => {
+                    if (extension.attributes[attribute] == undefined) {
+                        return;
+                    }
+                    const accessor = ArrayItem.Get(`${extensionContext}/attributes/${attribute}`, this._loader.gltf.accessors, extension.attributes[attribute]);
+                    promises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attribute] = { buffer : data, accessor : accessor };
+                    }));
+                };
+                const attributes = ["TRANSLATION", "ROTATION", "SCALE"];
+                for (let i = 0; i < attributes.length; ++i) {
+                    loadAttribute(attributes[i]);
+                }
+
+                return Promise.all(promises).then(() => {
+                    const instanceCount = attributeBuffers[attributes[0]].accessor.count;
+                    const digitLength = instanceCount.toString().length;
+                    const padNumber = function(num: number, length: number) {
+                        var str = String(num);
+                        while (str.length < (length)) { str = "0" + str; }
+                        return str;
+                    };
+
+                    for (let i = 0; i < instanceCount; ++i) {
+                        if (node._primitiveBabylonMeshes) {
+                            for (let j = 0; j < node._primitiveBabylonMeshes.length; ++j) {
+                                const babylonMeshPrimitive = node._primitiveBabylonMeshes[j];
+                                const instance = babylonMeshPrimitive.clone((babylonMeshPrimitive.name || babylonMeshPrimitive.id) + "_" + padNumber(i, digitLength), babylonTransformNode, true);
```

Comment:
Cloning is not the same as creating a new instance? `Mesh.createInstance`?

Reply:
ah, you're absolutely right here. I'm deep copying the meshes instead of instantiating them.